src/Node/Internal.hs

```diff
@@ -1141,32 +1176,38 @@ spawnHandler stateVar provenance action =
                       _nodeStateOutboundBidirectional = Map.alter alteration peer (_nodeStateOutboundBidirectional nodeState)
                     }
                     where
-                    alteration Nothing = Just $ Map.singleton nonce (someHandler, channelIn, peerDataVar, connBundle, False)
-                    alteration (Just map) = Just $ Map.insert nonce (someHandler, channelIn, peerDataVar, connBundle, False) map
+                    alteration Nothing = Just $ Map.singleton nonce (someHandler, channelIn, incrBytes, peerDataVar, connBundle, False)
+                    alteration (Just map) = Just $ Map.insert nonce (someHandler, channelIn, incrBytes, peerDataVar, connBundle, False) map
                 Local peer Nothing -> nodeState {
                       _nodeStateOutboundUnidirectional = Map.insert peer someHandler (_nodeStateOutboundUnidirectional nodeState)
                     }
 
+            incrBytes !n = modifySharedAtomic stateVar $ \nodeState -> do
+                let statistics' = stIncrBytes (handlerProvenancePeer provenance) n (_nodeStateStatistics nodeState)
+                modifySharedAtomic totalBytes $ \(!m) -> return (m + n, ())
+                return (nodeState { _nodeStateStatistics = statistics' }, ())
```

Comment:
Yes, that might become a point of contention.

As you suggested on slack, I'll change it so that the `Map` itself holds `MVar`s that can be modified each, without holding the `MVar` for the whole `Map`.